benchmarks/src/main/kotlin/chat/BenchmarkConfiguration.kt

```diff
@@ -0,0 +1,192 @@
+ */
+const val BENCHMARK_CONFIGURATION_FIELDS = "threads,userCount,maxFriendsPercentage,channelType,averageWork,benchmarkMode,dispatcherType,avgSentMessages,stdSentMessages,avgReceivedMessages,stdReceivedMessages"
+
+/**
+ * Class containing all the benchmark configuration and metrics of the run benchmark
+ */
+class BenchmarkConfiguration(
+        var threads: Int,
+        var users: Int,
+        var maxFriendsPercentage: Double,
+        var channelType: ChannelType,
+        var averageWork: Int,
+        var benchmarkMode: BenchmarkModes,
+        var dispatcherType: DispatcherTypes
+) {
+    val sentMessagesPerRun = ArrayList<Double>()
+
+    val receivedMessagesPerRun = ArrayList<Double>()
+
+    override fun toString(): String {
+        return "BenchmarkConfiguration(threads=$threads, users=$users, maxFriendsPercentage=$maxFriendsPercentage, channelType=$channelType, averageWork=$averageWork, benchmarkMode=$benchmarkMode, dispatcherType=$dispatcherType, sentMessagesPerRun=$sentMessagesPerRun, receivedMessagesPerRun=$receivedMessagesPerRun)"
+    }
+
+    fun configurationToString() : String {
+        return "[threads=$threads, users=$users, maxFriendsPercentage=$maxFriendsPercentage, channelType=$channelType, averageWork=$averageWork, benchmarkMode=$benchmarkMode, dispatcherType=$dispatcherType]"
+    }
+
+    fun toCSV(): String {
+        val standardDeviation = StandardDeviation()
+        val avgSentMessages = sentMessagesPerRun.average()
+        val avgReceivedMessages = receivedMessagesPerRun.average()
+        val stdSentMessages = standardDeviation.evaluate(sentMessagesPerRun.toDoubleArray())
+        val stdReceivedMessages = standardDeviation.evaluate(receivedMessagesPerRun.toDoubleArray())
+        return "$threads,$users,$maxFriendsPercentage,$channelType,$averageWork,$benchmarkMode,$dispatcherType,%.2f,%.2f,%.2f,%.2f".format(avgSentMessages, stdSentMessages, avgReceivedMessages, stdReceivedMessages)
+    }
+
+    fun toArray() : Array<String> {
+        return arrayOf(threads.toString(), users.toString(), maxFriendsPercentage.toString(), channelType.toString(), averageWork.toString(), benchmarkMode.toString(), dispatcherType.toString())
+    }
+
+    companion object {
+        fun parseArray(array : Array<String>) : BenchmarkConfiguration {
+            val threads = array[0].toInt()
+            val userCount = array[1].toInt()
+            val maxFriendsPercentage = array[2].toDouble()
+            val channelType = ChannelType.valueOf(array[3])
+            val averageWork = array[4].toInt()
+            val benchmarkMode = BenchmarkModes.valueOf(array[5])
+            val dispatcherType = DispatcherTypes.valueOf(array[6])
+            return BenchmarkConfiguration(threads, userCount, maxFriendsPercentage, channelType, averageWork, benchmarkMode, dispatcherType)
+        }
+
+        fun defaultConfiguration() : BenchmarkConfiguration {
```

Comment:
Why do you need this?

Reply:
Sometimes it is useful to test the benchmark on one particular configuration. This method was added for convenience.